apps/studio/data/database-roles/database-role-create-mutation.ts

```diff
@@ -0,0 +1,73 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { toast } from 'react-hot-toast'
+
+import { post } from 'data/fetchers'
+import { ResponseError } from 'types'
+import { databaseRolesKeys } from './keys'
+
+export type DatabaseRoleCreateVariables = {
+  projectRef: string
+  connectionString?: string
+  payload: {
+    name: string
+    is_superuser: boolean
+    can_login: boolean
+    can_create_role: boolean
+    can_create_db: boolean
+    is_replication_role: boolean
+    can_bypass_rls: boolean
+  }
```

Comment:
Similar to https://github.com/supabase/supabase/pull/19723#discussion_r1427466166